mixin/thanos/alerts/ruler.libsonnet

```diff
@@ -0,0 +1,121 @@
+{
+  local thanos = self,
+  ruler+:: {
+    jobPrefix: error 'must provide job prefix for Thanos Ruler alerts',
+    selector: error 'must provide selector for Thanos Ruler alerts',
+  },
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'thanos-ruler.rules',
+        rules: [
+          {
+            alert: 'ThanosRulerQueueIsDroppingAlerts',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to queue alerts.',
+            },
+            expr: |||
+              sum by (job) (thanos_alert_queue_alerts_dropped_total{%(selector)s}) > 0
+            ||| % thanos.ruler,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+          },
+          {
+            alert: 'ThanosRulerSenderIsFailingAlerts',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to send alerts to alertmanager.',
+            },
+            expr: |||
+              sum by (job) (thanos_alert_sender_alerts_dropped_total{%(selector)s}) > 0
+            ||| % thanos.ruler,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+          },
+          {
+            alert: 'ThanosRulerHighRuleExaluationFailures',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to evaluate rules.',
+            },
+            expr: |||
+              (
+                sum by (job) (rate(prometheus_rule_evaluation_failures_total{%(selector)s}[5m]))
+              /
+                sum by (job) (rate(prometheus_rule_evaluations_total{%(selector)s}[5m]))
+              * 100 > 5
+              )
+            ||| % thanos.ruler,
+
+            'for': '5m',
+            labels: {
+              severity: 'warning',
+            },
+          },
+          {
+            alert: 'ThanosRulerHighRuleExaluationWarnings',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} has high number of evaluation warnings.',
+            },
+            expr: |||
+              sum by (job) (rate(thanos_rule_evaluation_with_warnings_total{%(selector)s}[5m])) > 0
+            ||| % thanos.ruler,
+
+            'for': '5m',
+            labels: {
+              severity: 'warning',
+            },
+          },
+          {
+            alert: 'ThanosRulerRuleEvaluationLatencyHigh',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}}/{{$labels.pod}} has higher evaluation latency than interval for {{$labels.rule_group}}.',
+            },
+            expr: |||
+              (
+                sum by (job, pod, rule_group) (prometheus_rule_group_last_duration_seconds{%(selector)s})
+              >
+                sum by (job, pod, rule_group) (prometheus_rule_group_interval_seconds{%(selector)s})
+              )
```

Comment:
I'm not too sure about this rule, as it's always evaluation for us, even though everything seems fine.

Reply:
This still indicates a problem though. Its severity set to `warning`. We might also give a coefficient for the interval but I think we should have this. 

With this alert, you can detect you have too many rules for a group for example. Or your queries take too much time. 

It's not very critical but I still see value in this.